### ISSUE TITLE
4.2.5, 4.2.6 = crash city

### ISSUE THREAD

**Issue by ghost** [deepskystacker/DSS#133]
This program crashes all the time now, and for the smallest things. I load a dark file, click on it. Crash. I load a light file, click on it. Crash! right-click, properties...crash!
Other problem it complains that the pictures are all different sizes and refuses to do anything, even though they were all taken with the same camera on the same night and all part of the same sequence of 15 second exposures. (Meade DSI II PRO)
32 bits, 64 bits, it makes no difference. crash, crash, crash crash. on a windows server 2012R2 DELL T3600 workstation system with 8Gb RAM
Also cannot see what you have taken, the image is completely black. You can play around with the sliders at the top right but things are very faint and you cannot see hardly anything...yet I can load the same image into FITS Liberator and see details quite easily.

Shame, used to be a nice usable program with version 3.x so sad to see this program go down the drain like this :(

**Comment by perdrix52** [comment]
This is the first time I've heard from you and you've never posted on the DeepSkyStacker support mailing list at groups.io.

Can I suggest that you post your problems with details there and upload some of the troublesome images to a file sharing site and share them with us.

**Comment by skymapper** [comment]
Ah, nevermind. I have a solution. I found I still had version 3.3.4 in an archive and that works perfectly, no crashes. I will keep using that instead.

**Comment by Zi7ar21** [comment]
I haven't had any problems, using a somewhat similar system with 9GB of RAM.

**Comment by perdrix52** [comment]
It would really help me if you were to make your images available to debug this.

David

**Comment by grytom** [comment]
I think it is a bug in CTFITSIO library, because it behaves the same (crashing) in other application using this library. Here is some example resolutions I craeted and tested:
4000×3499 OK
4000×3500 crash
4000×3501 crash
4000×3502 crash
4000×3503 crash
4000×3504 OK
4000×3505 OK
4000×3506 OK
4000×3507 crash
4000×3508 OK
4000×3509 OK
4000×3510 OK
4000×3511 crash
4000×3512 OK
4000×3513 OK
4000×3514 crash
and so on. It doesn't matter if the file is 8bit, 16bit or more. The resolution is the problem. So if you are unlucky and your files has "crashing" resolution (like me, 4656×3520), it is not problem that DSS could resolve.

**Comment by perdrix52** [comment]
Ouch.   Are you able to open a problem report with the CFITSIO team?  Can you join the discussion in the deepskystacker mailing list on groups.io?   I appears another user has a similar issue (thread "4.2.6 is garbage").

**Comment by grytom** [comment]
Here is one of my files: [test_fit.zip](http://grytom.g6.cz/odkladiste/test_fit.zip)
I have my own application for capturing, so the header is as simple as possible.
I wrote a simple program to increase the height of the image by 1px and fill the "new" row with zeros. Those "enhanced" files works normally.

**Comment by perdrix52** [comment]
Interesting - I don't get a crash - what I do get is:

![image](https://user-images.githubusercontent.com/20211762/122742339-c010bf00-d27d-11eb-9008-6112043e2531.png)

**Comment by grytom** [comment]
I got exactly the same error and it crashes imediatelly after I click OK.
But this file is problematic in every FITS viewer (except my own ;-) ). Increasing height by 1px and filling the new row with zeros makes it work without any issues. Nothing else changes.

**Comment by perdrix52** [comment]
Have you considered that you may be creating mal-formed FITS files?   I’ve generally only seen that error when there was a genuine problem with the file.

 

I don’t know if there’s a tool to check that a FITS file is 100% OK?

 

D.

**Comment by grytom** [comment]
Header should be fine, problem is reading binary data after.
I downloaded application [fv from NASA](https://heasarc.gsfc.nasa.gov/docs/software/ftools/fv/) and it reads the header with all the information. Problem is reading binary data after. The weirdest thing is, if you try open the image several times, it eventually opens like there is nothing wrong.
File enhanced by 1px has no issues what so ever... when I wrote my app, I tested it in DSS to be sure it will work.
